### PR TITLE
docs: record DreamerV3 import boundary no-go (#1190)

### LINKED ISSUE #1190: research: probe DreamerV3 checkpoint import boundary for BR-08 gate
## Goal / Problem

**Evidence grade:** proposal

**Research question**
Can Robot SF reuse a clean DreamerV3 checkpoint/import boundary to test a small BR-08 pretraining-style warm start without forking RLlib or expanding the model-data contract prematurely?

**Hypothesis**
If RLlib exposes a stable world-model or RLModule checkpoint boundary, then a short imported-vs-scratch BR-08 gate comparison can show whether a warm-start path is worth pursuing before any larger pretraining program is opened.

**Scientific motivation**
Closed issue `#782` explicitly recommended a follow-up implementation issue called `DreamerV3 checkpoint import boundary probe for BR-08 gate`, but that follow-up was not created. The repository already has BR-08 gate/full configs and a runbook, so the smallest research next step is to probe the import boundary directly instead of reopening the broader design discussion.

## Scope

- Independent variables:
  - initialization mode: imported checkpoint weights vs from-scratch gate run.
- Dependent variables:
  - `eval/success_rate`
  - `eval/collision_rate`
  - `episode_return_mean`
- Control variables:
  - gate config, seed family, training steps, observation contract, and evaluation path.
- In scope:
  - identify the exact checkpoint tensors/modules that would need to transfer,
  - prove whether import into a fresh BR-08 gate run is clean,
  - run one bounded imported-vs-scratch gate comparison if the boundary is clean,
  - fail closed and stop if the path requires RLlib-specific checkpoint surgery or a fork.
- Out of scope:
  - a full offline pretraining stack,
  - changing benchmark claims,
  - widening the effort into a larger Dreamer redesign.

## Added Value Estimation

- Research value: high — answers the smallest actionable follow-up from `#782`.
- Benchmark value: medium — clarifies whether any warm-start path can remain provenance-safe.
- Publication value: medium — either yields a bounded positive signal or a justified no-go.
- Why now: the BR-08 gate/full configs and runbooks already exist, so the next unknown is the checkpoint boundary itself.

## Effort Estimation

- Rough estimate (hours): 6-12
- Best estimate (hours): 8
- Unknowns:
  - whether RLlib exposes a clean import surface,
  - whether a 100k-step gate comparison is enough to show a directional signal.

## Complexity Estimation

- Experimental complexity: medium
- Dependencies:
  - `#782`
  - `#578`
  - `#608`
  - `#609`
  - `configs/training/rllib_dreamerv3/benchmark_socnav_grid_br08_gate.yaml`
  - `scripts/training/train_dreamerv3_rllib.py`
- Open questions:
  - where exactly the reusable world-model boundary lives in the checkpoint,
  - whether the import path can stay clean without local RLlib modifications.

## Risk Assessment

- Statistical risk: medium — a short gate run may be noisy.
- Reproducibility risk: medium — checkpoint import behavior must stay explicit and versioned.
- Runtime risk: medium — even the gate probe needs meaningful RLlib runtime.
- Mitigation: keep the comparison bounded, use the existing gate config, and stop immediately if the import boundary is not clean.

## Affected Files

- `docs/context/issue_782_dreamerv3_pretraining_design.md` - design note that recommends this follow-up.
- `configs/training/rllib_dreamerv3/benchmark_socnav_grid_br08_gate.yaml` - canonical gate config.
- `scripts/training/train_dreamerv3_rllib.py` - launcher surface for the probe.
- `docs/training/dreamerv3_br08_slurm_handoff.md` - existing BR-08 runbook to extend with the probe path if warranted.

## Definition of Done

- [ ] The checkpoint boundary is inspected and the exact transferable module/tensor set is documented.
- [ ] One imported-vs-scratch BR-08 gate comparison is defined and run only if the boundary is clean.
- [ ] The outcome is recorded as either a measured improvement or a fail-closed no-action recommendation.
- [ ] The probe remains reproducible from a single canonical command/config path.

## Success Metrics

- The experiment can be rerun from a documented command.
- The result is reproducible across repeated runs with the same seed family.
- The analysis output directly answers whether the warm-start path is worth further work.

## Validation / Testing

- [ ] Use fixed seeds and the canonical BR-08 gate config.
- [ ] Document the exact environment and RLlib version assumptions.
- [ ] Run from `scripts/training/train_dreamerv3_rllib.py --config configs/training/rllib_dreamerv3/benchmark_socnav_grid_br08_gate.yaml`.
- [ ] Record the result in a checked-in evidence/context note.

## Estimate Discussion

- Why these values were proposed:
  - this is intentionally smaller than a full pretraining effort and focuses on the cleanest import-boundary question first.
- Uncertainty / confidence:
  - medium confidence.
- What evidence would move the estimate:
  - a clearly exposed RLlib import boundary would reduce effort; any need for checkpoint surgery or a fork should turn the result into a no-go quickly.

## Project Metadata

- Priority: Medium
- Effort (h): 8
- Reviewed: 2026-05-13

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Records the #1190 DreamerV3 checkpoint import-boundary probe as a fail-closed no-go: Ray/RLlib 2.53.0 exposes full Algorithm/RLModule restore, not a stable Robot SF world-model import contract.

## Linked Issues
- Closes #1190
- Relates to #782

## What Changed
- Added `docs/context/issue_1190_dreamerv3_checkpoint_import_boundary.md` with the inspected Ray/RLlib boundary, stop condition, artifact decision, and reopen criteria.
- Linked the note from `docs/README.md`, `docs/context/README.md`, and the #782 DreamerV3 pretraining design note.
- Added `tests/docs/test_dreamerv3_checkpoint_import_boundary.py` to guard the fail-closed decision and discoverability links.
- Included Ruff formatting fixes in `robot_sf/manual_control/__init__.py` and `tests/test_manual_control_recording.py` that the repo formatter applied on the current main snapshot.

## Why It Matters
- Added value: answers the smallest actionable follow-up from #782 without widening into a DreamerV3 redesign.
- Expected impact: prevents an imported-vs-scratch BR-08 run from being treated as benchmark evidence when the import path would require unversioned RLlib checkpoint surgery.
- Why this is worth merging now: it closes the open research question with a provenance-safe stop condition and clear reopen criteria.

## Validation / Proof
- Commands run:
  - `uv run --extra rllib python - <<PY ...` to inspect Ray/RLlib 2.53.0 DreamerV3 restore and checkpointable-component surfaces.
  - `uv run pytest tests/docs/test_dreamerv3_checkpoint_import_boundary.py -q` -> `2 passed`.
  - `git fetch origin main && git merge origin/main` -> already up to date.
  - `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main scripts/dev/pr_ready_check.sh` -> `3485 passed, 17 skipped, 3 warnings`; changed-file coverage `robot_sf/manual_control/__init__.py: 100.0%`; no TODO docstrings.
  - `uv run python scripts/dev/pr_ready_freshness.py write --base-ref origin/main` and `status --base-ref origin/main` -> fresh for `1e3e823fc70fcc8db097d285894c80eaae7e2cfe`.
- Evidence that the change works here: the docs guard verifies the note, docs index link, context index link, and #782 back-reference.
- Benchmarks or smoke tests, if applicable: no BR-08 gate comparison was launched because #1190 explicitly says to stop if the path requires RLlib-specific checkpoint surgery.

## Risks / Rollout
- Compatibility risks: none expected; this is docs/test coverage plus formatter-only Python cleanup.
- Failure modes: future DreamerV3 work may still need a durable checkpoint pointer and a config-first import API before any gate comparison is valid.
- Rollback or fallback plan: revert the docs note and docs-test if the project later adopts a clean RLlib-supported or Robot-SF-owned import boundary.

## Docs / Provenance
- Updated docs: `docs/context/issue_1190_dreamerv3_checkpoint_import_boundary.md`, `docs/context/README.md`, `docs/README.md`, `docs/context/issue_782_dreamerv3_pretraining_design.md`.
- Relevant design or provenance notes: `docs/context/issue_782_dreamerv3_pretraining_design.md`, `docs/context/dreamerv3_program_close_out_2026_04_30.md`, `docs/training/dreamerv3_rllib_drive_state_rays.md`.
- Any assumptions that need to be preserved: Ray/RLlib version inspected was 2.53.0; proof is inspection-level, not DreamerV3 performance evidence.
- Artifact decision: readiness refreshed ignored `output/coverage/` and `output/validation/pr_ready/`; these are disposable local validation artifacts and were not committed. No `output/dreamerv3/` model artifact was produced or made a downstream dependency.

## Follow-Up Issues
- Deferred work: only reopen or create follow-up work if a retained source DreamerV3 checkpoint, a durable artifact pointer, and a clean world-model/RLModule initialization API exist.
- Issues opened for follow-up: none; #1190 resolves as fail-closed no-action.

## Reviewer Notes
- Anything a reviewer should verify closely: the stop condition in the new note matches #1190’s instruction to fail closed when the path requires RLlib-specific checkpoint surgery.
- Any known limitations: no training comparison was run; this PR should not be cited as a DreamerV3 benchmark result.

### CODE REVIEW

File: docs/context/issue_1190_dreamerv3_checkpoint_import_boundary.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Issue 1190 DreamerV3 checkpoint import boundary probe
+
+Related issue: [#1190](https://github.com/ll7/robot_sf_ll7/issues/1190)
+Related design note: [issue_782_dreamerv3_pretraining_design.md](issue_782_dreamerv3_pretraining_design.md)
+Program boundary: [dreamerv3_program_close_out_2026_04_30.md](dreamerv3_program_close_out_2026_04_30.md)
+Runbook: [docs/training/dreamerv3_rllib_drive_state_rays.md](../training/dreamerv3_rllib_drive_state_rays.md)
```
[the remark this answers is not in the document]
Comment (ll7) [reply docs/context/issue_1190_dreamerv3_checkpoint_import_boundary.md]: Fixed on the branch head: the Runbook link text now matches the neighboring header links and uses the bare filename rather than a directory-prefixed label.

**Comment by ll7** [comment]
Synced `origin/main` into this draft PR and resolved the only merge conflict in the shared manual-control export list.

Local proof on the merge head:
- `uv run --active ruff check .`
- `uv run --active pytest tests/docs/test_dreamerv3_checkpoint_import_boundary.py -q`

This keeps the PR in the original docs-note scope: the DreamerV3 fail-closed boundary note and its docs-link guard remain the substantive diff, while the merge only absorbs current mainline manual-control changes so hosted checks can rerun on a clean base.
